Source/Core/VideoCommon/VertexShaderGen.cpp

```diff
@@ -498,6 +498,11 @@ ShaderCode GenerateVertexShaderCode(APIType api_type, const ShaderHostConfig& ho
                 "o.clipDist1 = clipDist1;\n");
     }
   }
+  else
+  {
+    // Same depth adjustment for Sonic. Unfortunately not just for clipping.
+    out.Write("o.pos.z = o.pos.z * (1.0 - 1e-7);\n");
```

Comment:
The "unfortunately not just for clipping" message doesn't entirely make sense to me.  Would something like "This hack needs to be applied even when depth control is not supported" make more sense?

Reply:
"unfortunately" because it affects depth for purposes other than clipping too.

Reply:
Alright, maybe something like "Unfortunately, this path will affect depth for purposes beyond clipping; there is no way to affect clipping only without support for backend_depth_clamp" then?

(Also, you mention that depth clamping coincides with clip control [here](https://bugs.dolphin-emu.org/issues/11897#note-2); does this hack work correctly if `backend_clip_control` is supported but `backend_depth_clamp` isn't or if `backend_depth_clamp` is supported but `backend_clip_control` isn't?)

Reply:
> you mention that depth clamping coincides with clip control

That may have been a mistake. I will check this again.

Reply:
I made a mistake. It is independent of clip control. Perhaps I was confusing it with clip distance.

Reply:
I updated the comment. I hope it is better.